### ISSUE TITLE
Use a bot to add a warning comment on issues older than a month

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#7459]
I missed this change and it threw me off a bit :)

I had an issue I was wanting to work on and last I remembered it was labeled as `ready to implement` but just before submitting a PR I noticed that the label was altered and now says `ask to implement`.

Was this change intentional?

I always read `ready to implement` as :
- anyone can start work on this and open a PR with a potential fix
- no further discussion or triage is needed in the issue

Whereas I read `ask to implement` as :
- no work should be done and opening a PR is not advised
- ask to be assigned to the issue first

-------

I don't have any strong opinions on this and I am fine either way but the new message seems a lot less welcoming.

Maybe I am misreading it?

**Comment by ybiquitous** [comment]
@romainmenke Yes, the label has been renamed recently. As I know, this discussion was a trigger: https://github.com/stylelint/stylelint/issues/4331#issuecomment-1865284953

Since some of `ready to implement` issues are very old, I think confirming the current status is good before tackling such an issue.

@jeddy3 Is my understanding correct?

---

Incidentally, we also need to update the maintainer guide mentioning `ready to implement`:

https://github.com/stylelint/stylelint/blob/5f5cada88393dcb27282a7868297e37a579e0293/docs/maintainer-guide/issues.md?plain=1#L16

**Comment by romainmenke** [comment]
Got it!
Thank you for clarifying 🙇 

Updating the maintainer guide would be nice because that is the first place I looked after noticing the change :)

**Comment by jeddy3** [comment]
> but the new message seems a lot less welcoming.

Yes, that's definitely one of the downsides of this change. Can we think of better wording?

Or, if anyone has time, turn on a bot that automatically adds a message to issues with the "ready to implement" label that are older than 1 month:

> "This issue is older than 1 month. Even though it is labelled as "ready to implement", please ask before opening a pull request as it may no longer be relevant.

That way we can return to the more welcoming message.

Either way, if an issue is recent, I think we can just follow our existing approach of simply switching the label to `wip` and opening a PR. There is no need to ask specifically.

**Comment by Mouvedia** [comment]
> Was this change intentional?
> 
> I always read `ready to implement` as :
> 
> * anyone can start work on this and open a PR with a potential fix
> * no further discussion or triage is needed in the issue

That's what I thought as well but I learned the hard way—twice—that it wasn't the case.
If you are interested in old issues with a lot of 👍 like I was then you could end up working on a PR to have it closed without being merged in the end. The reason being that the old issues are not being re-checked regularly in regard to new decisions that were taken since their last labelling. Few people make big contributions for old issues like I do.

tl;dr: realistically, it shouldn't affect anyone but me, nonetheless better be safe than sorry.

**Comment by jeddy3** [comment]
> Or, if anyone has time, turn on a bot that automatically adds a message to issues with the "ready to implement" label that are older than 1 month:
>> "This issue is older than 1 month. Even though it is labelled as "ready to implement", please ask before opening a pull request as it may no longer be relevant.

I've labelled the issue for this as it feels like the optimal solution.

**Comment by ybiquitous** [comment]
Finally, we've added a GitHub action for this automation and updated old issues!

See the action history:
https://github.com/stylelint/stylelint/actions/workflows/commenting-on-old-issues.yml

Also, see old issues with the "ask to implement" label:
https://github.com/stylelint/stylelint/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+ask+to+implement%22
